### DISCUSSION TITLE
Testnet Functionality

### DISCUSSION THREAD

**Discussion by ghost** [tracer-protocol/perpetual-pools-contracts#253] [Ideas]
@greenbookwebb I believe a more sophisticated testnet would be a valuable resource to the finengg simulation and observation process. I have flagged this in the issues backlog, but the requirements have not been fleshed out. 

Can you provide some insight into what would be achieved by a V2 testnet in the following states: 
(a) Minimum functionality. What is an absolute must have.  
(b) Ideal functionality. What would make the process more seamless and efficient without being excessive. 
(c) Nice to haves. If resources were unlimited, how would you use a testnet? 

Details should include: Test token faucet, oracle functionality, scheduling or executing trade volumes, output logs etc

If I've made any incorrect assumptions here please correct them.

**Comment by ghost** [comment]
Linked to user story #243 (creating a ledger of trades to be executed in order), user story #229 (oracle manipulation to simulate crash etc)

**Comment by greenbookwebb** [comment]
Hi, 

**Minimum Functionality:** 
- Having the contracts deployed. 
- We don't even need users. We just need some scripted bots interacting with the contracts. I'm unsure if this is easier or harder than just having users interacting with the contracts.
- output logs of performance. I can spell out further detail, if you'd like. All details would be great, really.

**Comment by greenbookwebb** [comment]
**Ideal Functionality**
- an interface where users can interact with.

**Comment by greenbookwebb** [comment]
Note that Alex has already pretty much created classes of actors (e.g. apes, spot farmers, etc) that are interactig with eachother in his simulations. 

I don't know if this could be helpful, or if the point of the testnet is to have real users interact with the pools. 

Perhaps we should have normal users, but we also have a spot farmer bot (because it's unlikely someone who we be doing this, naturally).

**Comment by ghost** [comment]
Moving to #285 discussion so we can keep it more comprehensive
